examples/error-destructure-binding-pattern/error_destructure_binding_pattern.bal

```diff
@@ -1,36 +1,42 @@
 import ballerina/io;
 
-type SampleError error<string, map<anydata|error>>;
+type SampleErrorData record {
+    string message?;
+    error cause?;
+    string info;
+    boolean fatal;
+};
+
+type SampleError error<string, SampleErrorData>;
 
 public function main() {
     string reason;
-    map<anydata|error> detail;
+    string info;
+    boolean fatal;
     // This error-destructure binding pattern will destructure an `error` value of the type `SampleError`
-    // and assign the values to two variable references as follows:
+    // and assign the values to relevant variable references.
     // The value of the reason string in the `SampleError` will be assigned to the variable `reason`.
-    // The value of the detail mapping will be assigned to the variable `detail`.
-    error(reason, detail) = getSampleError();
+    // The values in the detail mapping will be assigned to the relevant variables.
+    error(reason, info = info, fatal = fatal) = getSampleError();
     io:println("Reason String: " + reason);
-    io:println(io:sprintf("Detail Mapping: %s", detail));
+    io:println("Info: ", info);
+    io:println("Fatal: ", fatal);
 
     string reasonTwo;
-    anydata|error detailTwo;
-    anydata|error fatal;
-
-    // The detail mapping can further be destructured into existing variable references.
-    error(reasonTwo, { detail: detailTwo, fatal }) = getSampleError();
-    io:println("Reason String: " + reasonTwo);
-    io:println(io:sprintf("Detail Mapping Field One: %s", detailTwo));
-    io:println(io:sprintf("Detail Mapping Field Two: %s", fatal));
-
+    map<anydata|error> params;
+    // The detail mapping can be destructured into an existing `map<anydata|error>`-typed variable by using a rest parameter.
+    error(reasonTwo, ...params) = getSampleError();
+    io:println("Reason String: ", reasonTwo);
+    io:println("Detail Map: ", params);
+    
     // The underscore '_' sign can be used to ignore either the reason string or the detail mapping.
```

Comment:
```suggestion
    // The underscore '_' sign can be used to ignore either the reason string or the detailed mapping.
```

Reply:
Error object has two fields named "reason" and "detail".